examples/upgradeable/upgrader/src/contract.rs

```diff
@@ -0,0 +1,54 @@
+/// Helper contract to perform upgrade+migrate or rollback+downgrade in a single
+/// transaction.
+use soroban_sdk::{contract, contractimpl, symbol_short, Address, BytesN, Env, Symbol, Val};
+use stellar_upgradeable::UpgradeableClient;
+
+pub const MIGRATE: Symbol = symbol_short!("migrate");
+pub const ROLLBACK: Symbol = symbol_short!("rollback");
+
+#[contract]
+pub struct Upgrader;
+
+#[contractimpl]
+impl Upgrader {
+    pub fn upgrade(
+        env: Env,
+        contract_address: Address,
+        operator: Address,
+        new_wasm_hash: BytesN<32>,
+    ) {
+        let contract_client = UpgradeableClient::new(&env, &contract_address);
+
+        contract_client.upgrade(&new_wasm_hash, &operator);
+    }
+
+    pub fn upgrade_and_migrate(
+        env: Env,
+        contract_address: Address,
+        operator: Address,
+        new_wasm_hash: BytesN<32>,
+        migration_data: soroban_sdk::Vec<Val>,
+    ) {
+        let contract_client = UpgradeableClient::new(&env, &contract_address);
+
+        contract_client.upgrade(&new_wasm_hash, &operator);
+        // The types of the arguments to the migrate function are unknown to this
+        // contract, so we need to call it with invoke_contract.
+        env.invoke_contract::<()>(&contract_address, &MIGRATE, migration_data);
+    }
+
+    pub fn rollback_and_downgrade(
+        env: Env,
+        contract_address: Address,
+        operator: Address,
+        old_wasm_hash: BytesN<32>,
+        rollback_data: soroban_sdk::Vec<Val>,
+    ) {
+        let contract_client = UpgradeableClient::new(&env, &contract_address);
+
+        // The types of the arguments to the migrate function are unknown to this
+        // contract, so we need to call it with invoke_contract.
+        env.invoke_contract::<()>(&contract_address, &ROLLBACK, rollback_data);
+        contract_client.upgrade(&old_wasm_hash, &operator);
```

Comment:
`contract_client.upgrade` is used in downgrade function. It kind of bothered me to see `upgrade()` is used within `downgrade()` implementation

Reply:
Yeah, I see. I also thought about it, but decided that there's no need to define another function that exactly the same as the first one, just with a different name. Maybe, it's better to rename this one to `rollback_and_upgrade`.